roomserver/version/version.go

```diff
@@ -15,84 +15,58 @@
-	EventFormat               EventFormatID
-	EnforceSigningKeyValidity bool
+	Supported bool
+	Stable    bool
 }
 
-var roomVersions = map[RoomVersionID]RoomVersionDescription{
-	RoomVersionV1: RoomVersionDescription{
-		Supported:                 true,
-		Stable:                    true,
-		StateResolution:           state.StateResolutionAlgorithmV1,
-		EventFormat:               EventFormatV1,
-		EnforceSigningKeyValidity: false,
+var roomVersions = map[gomatrixserverlib.RoomVersion]RoomVersionDescription{
+	gomatrixserverlib.RoomVersionV1: RoomVersionDescription{
+		Supported: true,
+		Stable:    true,
 	},
-	RoomVersionV2: RoomVersionDescription{
-		Supported:                 false,
-		Stable:                    true,
-		StateResolution:           state.StateResolutionAlgorithmV2,
-		EventFormat:               EventFormatV1,
-		EnforceSigningKeyValidity: false,
+	gomatrixserverlib.RoomVersionV2: RoomVersionDescription{
+		Supported: false,
+		Stable:    false,
 	},
-	RoomVersionV3: RoomVersionDescription{
-		Supported:                 false,
-		Stable:                    true,
-		StateResolution:           state.StateResolutionAlgorithmV2,
-		EventFormat:               EventFormatV2,
-		EnforceSigningKeyValidity: false,
+	gomatrixserverlib.RoomVersionV3: RoomVersionDescription{
+		Supported: false,
+		Stable:    false,
 	},
-	RoomVersionV4: RoomVersionDescription{
-		Supported:                 false,
-		Stable:                    true,
-		StateResolution:           state.StateResolutionAlgorithmV2,
-		EventFormat:               EventFormatV3,
-		EnforceSigningKeyValidity: false,
+	gomatrixserverlib.RoomVersionV4: RoomVersionDescription{
+		Supported: false,
+		Stable:    false,
 	},
-	RoomVersionV5: RoomVersionDescription{
-		Supported:                 false,
-		Stable:                    true,
-		StateResolution:           state.StateResolutionAlgorithmV2,
-		EventFormat:               EventFormatV3,
-		EnforceSigningKeyValidity: true,
+	gomatrixserverlib.RoomVersionV5: RoomVersionDescription{
+		Supported: false,
+		Stable:    false,
 	},
 }
 
-func GetDefaultRoomVersion() RoomVersionID {
-	return RoomVersionV1
+// GetDefaultRoomVersion contains the room version that will, by
+// default, be used to create new rooms on this server.
+func GetDefaultRoomVersion() gomatrixserverlib.RoomVersion {
+	return gomatrixserverlib.RoomVersionV1
 }
 
-func GetRoomVersions() map[RoomVersionID]RoomVersionDescription {
+// GetRoomVersions returns a map of all known room versions to this
+// server.
+func GetRoomVersions() map[gomatrixserverlib.RoomVersion]RoomVersionDescription {
 	return roomVersions
 }
 
-func GetSupportedRoomVersions() map[RoomVersionID]RoomVersionDescription {
-	versions := make(map[RoomVersionID]RoomVersionDescription)
+// GetSupportedRoomVersions returns a map of descriptions for room
+// versions that are supported by this homeserver.
+func GetSupportedRoomVersions() map[gomatrixserverlib.RoomVersion]RoomVersionDescription {
```

Comment:
Nit: I'd prefer if we drop the `Get` as it isn't a common Go idiom - https://golang.org/doc/effective_go.html#Getters

Reply:
Will do.